sim-lib/src/lib.rs

```diff
@@ -670,12 +684,17 @@ async fn produce_simulation_results(
                 match output{
                     Some(simulation_output) => {
                         match simulation_output{
-                            SimulationOutput::PaymentSent(dispatched_payment) => {
+                            SimulationOutput::SendPaymentSuccess(dispatched_payment) => {
                                 let source_node = nodes.get(&dispatched_payment.source).unwrap().clone();
                                 set.spawn(track_payment_result(
-                                    source_node,results.clone(),simulation_output, shutdown.clone(),
+                                    source_node,results.clone(),dispatched_payment, shutdown.clone(),
                                 ));
                             },
+                            SimulationOutput::SendPaymentFailure(payment, result) => {
+                                if results.clone().send((payment, result)).await.is_err() {
+                                    log::debug!("Could not send payment result");
+                                }
```

Comment:
made the same note

Reply:
> This is pre-existing, but I think we should break if we can't send into the results channel? Because something has exit downstream.

I think we should only break out of these listeners if shutdown is called. If something exits with finality, it should trigger shutdown, which leads to break(s), which can lead to channel send fail (because we bias to shutdowns?), but we log these send failures like we do here